### PR TITLE
Implementing additionalProperties to Schema and Boolean

### PR DESCRIPTION
**Pull request by gracekarina**
refs swagger-api/swagger-core#2461 - swagger-api/swagger-parser#499

### CODE REVIEW

File: src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java [hunk 1]
```diff
@@ -152,10 +152,54 @@ public void verifyGetEmailResponse() throws Exception {
...
+    /**
+     * verifies that the email return value is automatically created
+     */
+    @org.junit.Test
+    public void verifyAdditionalPropertyResponse() throws Exception {
+        Map<String, String> queryParams = new HashMap<String, String>();
+
+        String str = client.invokeAPI("/mockResponses/additionalPropertiesTest", "GET", queryParams, null, new HashMap<String, String>(), null, "application/json", null, new String[0]);
+        assertEquals(str, "{\"key\":\"key\",\"value\":\"string\"}");
+    }
```
Comment (frantuma) [inline src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: doesn't seem right, should be something like `"{\"foo\":0,\"additionalPropertyExampleKey\":\"string\"}"`

File: src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java [hunk 2]
```diff
@@ -152,10 +152,54 @@ public void verifyGetEmailResponse() throws Exception {
...
+    /**
+     * verifies that the email return value is automatically created
+     */
+    @org.junit.Test
+    public void verifyAdditionalPropertyIntegerResponse() throws Exception {
+        Map<String, String> queryParams = new HashMap<String, String>();
+
+        String str = client.invokeAPI("/mockResponses/additionalProperties1", "GET", queryParams, null, new HashMap<String, String>(), null, "application/json", null, new String[0]);
+        assertEquals(str, "{\"key\":\"key\",\"value\":{\"bar\":0}}");
+    }
```
Comment (frantuma) [inline src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: doesn't seem right, should be something like `"{\"foo\":0,\"additionalPropertyExampleKey\":{\"bar\":0}}}"`

File: src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java [hunk 3]
```diff
@@ -152,10 +152,54 @@ public void verifyGetEmailResponse() throws Exception {
...
+    /**
+     * verifies that the email return value is automatically created
+     */
+    @org.junit.Test
+    public void verifyBadAdditionalPropertyResponse() throws Exception {
+        Map<String, String> queryParams = new HashMap<String, String>();
+
+        String str = client.invokeAPI("/mockResponses/badAdditionalProperties", "GET", queryParams, null, new HashMap<String, String>(), null, "application/json", null, new String[0]);
+        assertEquals(str, "{\"foo\":0}");
+    }
```
Comment (frantuma) [inline src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: why "Bad"? change to "False" or something.

Comment (gracekarina) [reply src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: changed

File: src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java (hunk 1 above)
Comment (gracekarina) [reply src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: I changed the code, and try not to break the older tests in the process

File: src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java (hunk 2 above)
Comment (gracekarina) [reply src/test/java/io/swagger/oas/test/integration/responses/PrimitiveResponseTestIT.java]: I changed the code, and try not to break the older tests in the process

**Pull request merged**